src/test/pmem_reorder_simple/pmem_reorder_simple.c

```diff
@@ -120,6 +124,9 @@ main(int argc, char *argv[])
 		memset(structp, 0, sizeof(*structp));
 
 	VALGRIND_LOG_STORES;
+	/* change only for VALGRIND_DEFAULT_REORDER test */
```

Comment:
The way I imagined this would be used is to have a scope of code that is critical/simple and should use specific type of reordering, for example:
```
  VALGRIND_FULL_REORDER; /* begin critical work scope */
  /* some critical things being done in pmem */ 
 VALGRIND_DEFAULT_ENGINE; /* end critical work scope */
```

In this test, I don't really understand what's the point of switching the reorder engine? We switch it to PARTIAL_REORDER, and the immediately switch back to the default engine inside of the write_ functions.